config/webpack/configuration.js

```diff
@@ -12,7 +12,7 @@ const devServer = safeLoad(readFileSync(join(configPath, 'development.server.yml
 
 // Compute public path based on environment and CDN_HOST in production
 const ifHasCDN = env.CDN_HOST !== undefined && env.NODE_ENV === 'production';
-const devServerUrl = `http://${devServer.host}:${devServer.port}/${paths.entry}/`;
+const devServerUrl = `http://${env.LOCAL_DOMAIN || devServer.host}:${devServer.port}/${paths.entry}/`;
```

Comment:
All the changes made in this PR are to facilitate running a development/testing environment across multiple servers/containers, rather than entirely on `localhost` directly. No new options were introduced, only adapted to use cases they weren't being used in before. I'll address your specific points one at a time, in order:

- **`development.server.yml`:** Since this is a file that is checked into Git, we'd have to replace it every time the code is updated upstream, and Git would complain about it every time, too. It's extra overhead that we don't have with `.env`.

- **`NODE_ENV` (and `RAILS_ENV`):** I considered adding another environment (`nanobox-dev` or some such) but decided that would quickly get out of hand with all sorts of additional environments being desired all over the map (including, in this case, at least the addition of `nanobox-test`). This feels like opening the door to some things we really don't want to have, not to mention needing code changes in a huge number of places throughout the codebase to account for the alternative environments everywhere they're consulted. I'd just set these to `production` if WebPack would continue working properly that way.

- **Extra environment variable:** This is actually not a new env var at all - it's the same one used in production to determine the domain used anywhere Mastodon needs to know it. It gives us an easy mechanism in development to differentiate between a standard single-server dev environment, and a more production-like multi-container dev environment. Single-server environments simply continue not setting it, while multi-server setups have the option of setting it (just as they would in production, albeit to a development value) to work properly in that setup.

I'm open to alternative approaches, but this is the approach with the least impact on other parts of the codebase, among the ones I could think of.